src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -177,20 +186,23 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 		}
 
 		$options = wp_parse_args( $options, array( 'with_supports' => true ) );
+		if ( null === static::$theme_json_data ) {
 
-		if ( null === static::$theme ) {
 			$theme_json_data = static::read_json_file( static::get_file_path_from_theme( 'theme.json' ) );
 			$theme_json_data = static::translate( $theme_json_data, wp_get_theme()->get( 'TextDomain' ) );
+
 			/**
 			 * Filters the data provided by the theme for global styles & settings.
 			 *
 			 * @since 6.1.0
 			 *
 			 * @param WP_Theme_JSON_Data Class to access and update the underlying data.
 			 */
-			$theme_json      = apply_filters( 'theme_json_theme', new WP_Theme_JSON_Data( $theme_json_data, 'theme' ) );
-			$theme_json_data = $theme_json->get_data();
-			static::$theme   = new WP_Theme_JSON( $theme_json_data );
+
+			static::$theme_json_data = $theme_json_data;
+			$theme_json              = apply_filters( 'theme_json_theme', new WP_Theme_JSON_Data( $theme_json_data, 'theme' ) );
+			$theme_json_data         = $theme_json->get_data();
```

Comment:
If we set `static::$theme_json_data` after these two lines. Then the theme json file does not apply.